client/src/game/shapes/shape.ts

```diff
@@ -339,6 +360,16 @@ export abstract class Shape {
         if (sync) socket.emit("Shape.Layer.Change", { uuid: this.uuid, layer, floor: newLayer.floor });
     }
 
+    rotate(degrees: number, sync: boolean): void {
+        const oldLayer = layerManager.getLayer(this.floor, this.layer);
+        this.rotationDegrees = degrees;
+        if (this.rotationDegrees >= 360) this.rotationDegrees = 0;
+        if (oldLayer === undefined) return;
+        oldLayer.invalidate(true);
```

Comment:
You can just call `this.invalidate` instead of having to fetch the layer yourself.

Reply:
done